public_html/wp-content/plugins/pattern-directory/src/pattern-post-type/details.js

```diff
@@ -1,33 +1,44 @@
 /**
- * External dependencies
+ * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { createElement, useEffect, useState } from '@wordpress/element';
+import { useEffect, useState } from '@wordpress/element';
 import { PluginDocumentSettingPanel } from '@wordpress/edit-post';
-import { TextControl, TextareaControl } from '@wordpress/components';
+import { ComboboxControl, TextControl, TextareaControl } from '@wordpress/components';
 import { useDispatch, useSelect } from '@wordpress/data';
 
+const localeData = window.wporgLocaleData || {};
+const localeOptions = [];
+for ( const [ key, value ] of Object.entries( localeData ) ) {
+	localeOptions.push( {
+		value: key,
+		label: value,
+	} );
+}
+
 const PatternDetails = () => {
 	const { editPost } = useDispatch( 'core/editor' );
 	const postMetaData = useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'meta' ) || {} );
 	const postTitle = useSelect( ( select ) => select( 'core/editor' ).getEditedPostAttribute( 'title' ) || '' );
 	const [ description, setDescription ] = useState( postMetaData.wpop_description );
+	const [ locale, setLocale ] = useState( postMetaData.wpop_locale );
 
 	useEffect( () => {
 		editPost( {
 			meta: {
 				...postMetaData,
 				wpop_description: description,
+				wpop_locale: locale,
 			},
 		} );
-	}, [ description ] );
+	}, [ description, locale ] );
```

Comment:
We removed the `useEffect` call here in #112, so you can set this using `editPost` in the `onChange` directly - you shouldn't need the local state `locale`.

Reply:
Fixed with 840e9cc